platform/wayland/cog-view-wl.c

```diff
@@ -214,8 +212,7 @@ cog_wl_view_create_backend(CogView *view)
 bool
 cog_wl_view_does_image_match_win_size(CogWlView *view)
 {
-    CogWlPlatform *platform = (CogWlPlatform *) cog_platform_get();
-    CogWlViewport *viewport = COG_WL_VIEWPORT(platform->viewport);
+    CogWlViewport *viewport = COG_WL_VIEWPORT(cog_view_get_viewport((CogView *) view));
```

Comment:
The `cog_view_get_viewport()` function returns a `(transfer full)` value with an added reference, so we have to unref after using it. The easiest way is to use `g_autoptr`:

```suggestion
    g_autoptr(CogWlViewport) viewport = COG_WL_VIEWPORT(cog_view_get_viewport((CogView *) view));
```